internal/resourceproviders/errors.go

```diff
@@ -33,7 +63,7 @@ Could suggest that the Resource Provider "Microsoft.Foo" requires registration,
 this could also indicate that this Azure Region doesn't support this API version.
 
 More information on the "resource_provider_registrations" property can be found here:
-https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#resource_provider_registrations
+https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#improved-resource-provider-registration
```

Comment:
The reporter of gh-27110 showed that the original anchored link was broken. Updated it to the next best link I could find. Happy to change if there's something more appropriate.